### ISSUE TITLE
BigSur disk doesn't appear after 1st reboot in bootloader

### ISSUE THREAD

**Issue by Moustik2002** [SpaceinvaderOne/Macinabox#44]
Hi, i hope this is not a dupplicate issue reported there i try to find an answer on my own without any sucessful result.

i'm following tutorial from space invader to create my own bigsur virtual machin through Unraid.

1st try yesterday, i manage to follwo method installation one which lead me to have working catalina instead of bigsur.
So i follow the recommendation linked in the issue closed #33 

So i manage to remove macinabox and all traces of it thanks to the Bpelech's comment, and redo install w/ method 2

and now i'm stuck here :

![Capture d’écran 2021-05-04 164412](https://user-images.githubusercontent.com/7987482/117021906-0f178a80-acf8-11eb-85b2-7b0143069f8a.png)

instead to have this :

![Capture d’écran 2021-05-04 164547](https://user-images.githubusercontent.com/7987482/117022135-45eda080-acf8-11eb-8a08-bfb3bd511c24.png)

from my research there maybe a link with BigSur-opencore.img wich could be damaged or else.

My question is i would like to be able to download it again but each time i tried, img file is back to isos folder instantely and i've no clue where to clean it up. 

Looking to log say : "BigSur-opencore.img already exists. If you want me to replace you will need to manually delete it first"
but no still no clue about folder location.

Anyway thx for your soft @SpaceinvaderOne and if you could help me it would be very appreciated !

Moustik !

**Comment by Moustik2002** [comment]
i'm not sure what's going on with my Big sur install i've tried to delete everything as indicated on spaceinvader video and get a hard reboot issue each time i run macos base system. I've also had an issue when starting VM initially saying "operation failed: unable to find any master var store for loader: /mnt/user/system/custom_ovmf/Macinabox_CODE-pure-efi.fd" than i get ride deleting "custom_ovf folder" before reinstalling macinabox.

In a nutshell i'm completely lost and i need assistance if @SpaceinvaderOne  as some spare time to look at my issue.

**Comment by SilentBob69** [comment]
The same thing happens to me. But I suspect that Apple has changed something. I was able to install Catalina without any problems. After that I did an update to BigSur. Here, too, the error appeared. 
By the way, the wrong network card is still displayed.

**Comment by Moustik2002** [comment]
thx for the feedback @SilentBob69  i feel less lonely w/ my issue :)

**Comment by plexecutor** [comment]
Just keep choosing MacOS Installer until it is done installing then you should see the disk when it is done with all steps

**Comment by Moustik2002** [comment]
@plexecutor  no MacOS Installer shown on bootloader :

![Capture d’écran 2021-05-14 155734](https://user-images.githubusercontent.com/7987482/118281124-30812f00-b4cd-11eb-8b6a-0e40285b1a8b.png)

**Comment by Daandeve** [comment]
If you use the OpenCore iso of the open pull request #47. It boots no problem

**Comment by Moustik2002** [comment]
@Daandeve  sure thing and thx for the guidance. Speaking of guidance i'm not really github fluent could you please tell me how to get opencore.img from the pull request ?

**Comment by Daandeve** [comment]
Sure, this is the direct link (easier)
https://github.com/SpaceinvaderOne/Macinabox/blob/c1d7ecbf4eb01029d98fdda29166391ba5077dc3/bootloader/OpenCore.img.zip
You can click download -> extract the iso -> Rename existing (for backup purposes) -> Name the new OpenCore iso the same as the old OpenCore iso

**Comment by Moustik2002** [comment]
Ok done, unfortunatelly VM looping reboot .

When i press enter on the boot loader :
![Capture d’écran 2021-05-14 155734](https://user-images.githubusercontent.com/7987482/118364383-8373e880-b598-11eb-9a13-be4c18177846.png)

Then it displays this for a moment :

![Capture d’écran 2021-05-15 161158](https://user-images.githubusercontent.com/7987482/118364418-9d153000-b598-11eb-8317-bea3d70364e4.png)

And then back to square 1 :'(

Do i have skipped something ?

**Comment by Daandeve** [comment]
The steps I described is what worked for me, I did a fresh install of the OS. You should have the disk that you named listed in the boot loader. But on your screenshot is the disk missing. Maybe your install corrupted??

**Comment by Moustik2002** [comment]
ok so i'm gonna try a clean macinabox install but keeping opencore.img somwhere because i giess it will reinstall the non working one instead.

**Comment by Moustik2002** [comment]
ok, same results w/ clean install 
![Capture d’écran 2021-05-15 173616](https://user-images.githubusercontent.com/7987482/118369603-4150a400-b5a4-11eb-9dc7-3b782d87147f.png)
Rebooting just after this screenshot; i'm completely lost. Does someone need some logs or else ? I'll be glad to provide if you tell me where to fetch them.

**Comment by cracyfloyd** [comment]
same here today, i see only macos base system partition. a installation is not possible inside the vm. no gpt partition available.

**Comment by Moustik2002** [comment]
thx @cracyfloyd  i fell a little bit less lonely :)

**Comment by Moustik2002** [comment]
anyone here to support this issue ?

**Comment by CanisHelix** [comment]
Same as @Moustik2002, stuck in a boot loop with the OpenCore image from the PR. Clean install so nothing previous to mess it up is present.

**Comment by 82ghost82** [comment]
To all that are having issues with bootloops.
Please have a look at this:
https://forums.unraid.net/topic/84601-support-spaceinvaderone-macinabox/?do=findComment&comment=1009972

**Comment by Moustik2002** [comment]
Hey Ghost82 thx for the link to the HUGE thread :)
I don't know if you will to continue the discussion here or there, Any way i tried to go back in time and tried the edition of Opencore.img (renamed as BigSur-opencore.img to keep path of ISO in VM conf) you provided and this happens :

modprobe nbd max_part=8 

->OK

qemu-nbd --connect=/dev/nbd0 /mnt/user/isos/BigSur-opencore.img
WARNING: Image format was not specified for '/mnt/user/isos/BigSur-opencore.img' and probing guessed raw.
         Automatically detecting the format is dangerous for raw images, write operations on block 0 will be restricted.
         Specify the 'raw' format explicitly to remove the restrictions.
qemu-nbd: Failed to set NBD socket
qemu-nbd: Disconnect client, due to: Failed to read request: Unexpected end-of-file before all bytes were read
--> Erf ! OK then let's specify it is RAW !

qemu-nbd --format=raw --connect=/dev/nbd0 /mnt/user/isos/BigSur-opencore.img
qemu-nbd: Failed to set NBD socket
qemu-nbd: Disconnect client, due to: Failed to read request: Unexpected end-of-file before all bytes were read

--> i guess i'm stuck. I also tried to mount it with windows explorer to see if i'm able to browse inside but with no result (unable to mount the image as well)

Any guess ? .

**Comment by 82ghost82** [comment]
@Moustik2002 
I have no issues with:

modprobe nbd max_part=8
cd to/path/where/OpenCore/image/
qemu-nbd --connect=/dev/nbd0 --format=raw OpenCore.img
mkdir /mount/point/
mount /dev/nbd0p1 /mount/point/

I think you need to disconnect after you tried to connect without the raw attribute, then reconnect again with raw.

**Comment by Moustik2002** [comment]
Indeed i didn't know i would have to disconnect it before trying again. So i've good news so far !

![image](https://user-images.githubusercontent.com/7987482/123677640-ab1bc900-d845-11eb-8b01-5fb431d56b40.png)

Soi've followed the guide line link below but i've added the "qemu-nbd --connect=/dev/nbd0 **--format=raw** " to the original opencore.img provided by Spaceinvader and edit the script as also described and things seem to be unlocked !

https://forums.unraid.net/topic/84601-support-spaceinvaderone-macinabox/?do=findComment&comment=1004897

Will let you posted w/ final result ! A huge thx to you @82ghost82  i owe you a beer !

**Comment by Moustik2002** [comment]
Yeah ! Finally
![image](https://user-images.githubusercontent.com/7987482/123693924-f2f81b80-d858-11eb-9885-1bef60aba12d.png)

**Comment by Dkgamga** [comment]
> Just keep choosing MacOS Installer until it is done installing then you should see the disk when it is done with all steps

This worked for me
